pennylane/ops/op_math/sum.py

```diff
@@ -458,3 +470,460 @@ def get_summands(self, cutoff=1.0e-12):
+                self.ops, grouping_type=grouping_type, method=method
+            )
+
+    @qml.QueuingManager.stop_recording()
+    def _simplify_coeffs_ops(self):
+        """Simplify coeffs and ops
+        
+        Returns:
+            coeffs, ops, pauli_rep"""
+
+        if len(self.ops) == 0:
+            return self
+
+        # try using pauli_rep:
+        if pr := self.pauli_rep:
+
+            wire_order = self.wires
+            if len(pr) == 0:
+                return [], [], pr
+
+            # collect coefficients and ops
+            coeffs = []
+            ops = []
+
+            for pw, coeff in pr.items():
+                pw_op = pw.operation(wire_order=wire_order)
+                ops.append(pw_op)
+                coeffs.append(coeff)
+
+            return coeffs, ops, pr
+
+        if len(self.ops) == 1:
+            return self.coeffs, [self.ops[0].simplify()], pr
+
+        op_as_sum = qml.sum(*self.operands)
+        op_as_sum = op_as_sum.simplify()
+        coeffs, ops = op_as_sum.terms()
+        return coeffs, ops, None
+
+    def simplify(self):
+        r"""TODO
+        """
+        coeffs, ops, pr = self._simplify_coeffs_ops()
+        return LinearCombination(coeffs, ops, _pauli_rep=pr)
+
+    def _obs_data(self):
+        r"""Extracts the data from a LinearCombination and serializes it in an order-independent fashion.
+
+        This allows for comparison between LinearCombinations that are equivalent, but are defined with terms and tensors
+        expressed in different orders. For example, `qml.PauliX(0) @ qml.PauliZ(1)` and
+        `qml.PauliZ(1) @ qml.PauliX(0)` are equivalent observables with different orderings.
+
+        .. Note::
+
+            In order to store the data from each term of the LinearCombination in an order-independent serialization,
+            we make use of sets. Note that all data contained within each term must be immutable, hence the use of
+            strings and frozensets.
+
+        **Example**
+
+        >>> H = qml.LinearCombination([1, 1], [qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(0)])
+        >>> print(H._obs_data())
+        {(1, frozenset({('PauliX', <Wires = [1]>, ()), ('PauliX', <Wires = [0]>, ())})),
+         (1, frozenset({('PauliZ', <Wires = [0]>, ())}))}
+        """
+        data = set()
+
+        coeffs_arr = qml.math.toarray(self.coeffs)
+        for co, op in zip(coeffs_arr, self.ops):
+            obs = op.non_identity_obs if isinstance(op, Tensor) else [op]
+            tensor = []
+            for ob in obs:
+                parameters = tuple(
+                    str(param) for param in ob.parameters
+                )  # Converts params into immutable type
+                if isinstance(ob, qml.GellMann):
+                    parameters += (ob.hyperparameters["index"],)
+                tensor.append((ob.name, ob.wires, parameters))
+            data.add((co, frozenset(tensor)))
+
+        return data
+
+    def compare(self, other):
```

Comment:
The origin of this is coming from `Observable` - should we continue to support this in `LinearCombination`?